### DISCUSSION TITLE
Samsung DeX mode/Lenovo Tablet Productivity mode support on the app

### DISCUSSION THREAD

**Discussion by goliv04053** [breezy-weather/breezy-weather#1189] [Ideas]
### Describe your idea

There are some apps in these modes who have full screen compatibility when using keyboard and mouse. I think that would be interesting to the app have support to this function when in this mode. 

### Other details

I use the app on my phone and on my tablet (Lenovo P11 Plus) and when I use my keyboard and mouse to do Productivity/office work I would like to have a Full screen like when I am not using this mode. Full screen works there.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.5 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
The app already runs in full screen, I don't understand what you're talking about?

**Reply by goliv04053** [reply]
When the productivity mode option is enabled on the Lenovo Tablet it doesn't go full screen. Only with the desktop mode disabled it permits to go on full screen. (I use one). I meant Breezy Weather have more compatibility with Productivity/Desktop mode.

**Reply by papjul** [reply]
Maybe contact Lenovo to ask them why? Works fine on my phone with nothing particular enabled

**Reply by goliv04053** [reply]
I said that the issue was with the app not going on full screen with Productivity mode feature on Lenovo Tablets enabled (the feature makes the Tablet run like Samsung DeX mode). The application works normally on full screen when the feature is not enabled how I said before.

**Reply by papjul** [reply]
So it's an issue with this Productivity mode, right? What does it have to do with us?

**Reply by goliv04053** [reply]
The question I asked is if it were possible to implement a compatibility with these function. There are some Google and even some mainstream apps who work native without any issues on the already cited.

**Reply by papjul** [reply]
Please tell us what we need to do

**Reply by goliv04053** [reply]
Okay. The main issue is:
- [ ] Full screen resizing not working

I took some screenshots.

![Screenshot_20240711-124035.jpg](https://github.com/breezy-weather/breezy-weather/assets/22873288/66645793-fde3-4892-b28f-5d670cb1da3d)

![Screenshot_20240711-124002.jpg](https://github.com/breezy-weather/breezy-weather/assets/22873288/bd1bbcf0-b59d-4f94-9107-17be69a7978e)

![Screenshot_20240711-124025.jpg](https://github.com/breezy-weather/breezy-weather/assets/22873288/5854e9a8-f287-4b5c-8edd-da9e1cb2fbad)

Also, before I begun to comment I gave a search if there isn't any direct documentation who could help with that on Lenovo side but for DeX counterpart It has better docs on the Samsung side. (Maybe It can help my situation even if it's for Samsung. I do not know)

https://developer.samsung.com/samsung-dex (Some info about Samsung DeX)

https://forums.lenovo.com/t5/Lenovo-Developer-Community/ct-p/Developer_Community (Lenovo Forum)

https://forums.lenovo.com/topic/view/1088/5068560

**Reply by papjul** [reply]
This is still not clear to me how it is related to us? The forum says the Productivity mode is a proprietary component by Lenovo. Those resize buttons you see don't belong to our app, so you have to contact Lenovo support to ask for a fullscreen button. Am I missing something?

**Reply by goliv04053** [reply]
I meant if it were possible to make the the Breezy Weather part be compatible with the desktop mode function.

**Reply by papjul** [reply]
How do I do that?

**Reply by goliv04053** [reply]
I.... do not know. I'm sorry for taking up your time. Your app is incredible. Lenovo doesn't have a good documentation about this function so.... how can something be made without the basis information to implement it? Well, have a good night. Here is 1 PM and I stayed all night awake and haven't slept yet. Don't forget to drink water.

**Reply by papjul** [reply]
If there is no documentation or if it requires a proprietary component, there's nothing we can do. You will have to contact Lenovo to ask for more technical information.
